### ISSUE TITLE
Trying to compile on Ubuntu 20.04

### ISSUE THREAD

**Issue by roracle** [muravjov/bombono-dvd#25]
I had to change a few things in the code.

> print 'bla blah blaw'

is actually

> print ('bla blah blaw') 

now days.

I also had to change has_key to __contains__

But then after all that, I get this output when I compile.

> scons: Reading SConscript files ...
Using config file: config.opts
scons: done reading SConscript files.
scons: Building targets ...
scons: building associated VariantDir targets: build
g++ -o build/src/mgui/project/mconstructor-main.o -c -Wall -W -Wno-reorder -O2 -ansi -std=c++11 -Wno-deprecated-declarations -DBOOST_SP_USE_QUICK_ALLOCATOR -DBOOST_SYSTEM_NO_DEPRECATED -DNDEBUG -DEXT_BOOST -Ibuild/src -Isrc -Ilibs/boost-logging -Ilibs/asl -Ibuild/src -Isrc -I/usr/include/glibmm-2.4 -I/usr/lib/x86_64-linux-gnu/glibmm-2.4/include -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/sigc++-2.0 -I/usr/lib/x86_64-linux-gnu/sigc++-2.0/include -I/usr/include/libxml++-2.6 -I/usr/lib/x86_64-linux-gnu/libxml++-2.6/include -I/usr/include/libxml2 -I/usr/include/x86_64-linux-gnu -I/usr/include/gtkmm-2.4 -I/usr/lib/x86_64-linux-gnu/gtkmm-2.4/include -I/usr/include/atkmm-1.6 -I/usr/include/gtk-unix-print-2.0 -I/usr/include/gtk-2.0 -I/usr/include/gdkmm-2.4 -I/usr/lib/x86_64-linux-gnu/gdkmm-2.4/include -I/usr/include/giomm-2.4 -I/usr/lib/x86_64-linux-gnu/giomm-2.4/include -I/usr/include/pangomm-1.4 -I/usr/lib/x86_64-linux-gnu/pangomm-1.4/include -I/usr/include/cairomm-1.0 -I/usr/lib/x86_64-linux-gnu/cairomm-1.0/include -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/harfbuzz -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/libpng16 src/mgui/project/mconstructor-main.cpp
In file included from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/project/mconstructor-main.cpp:22:
/usr/include/glibmm-2.4/glibmm/helperlist.h: In instantiation of ‘void Glib::HelperList<T_Child, T_CppElement, T_Iterator>::erase(Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator, Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator) [with T_Child = Gtk::Table_Helpers::Child; T_CppElement = Gtk::Widget; T_Iterator = Glib::List_Iterator<Gtk::Table_Helpers::Child>; Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator = Glib::List_Iterator<Gtk::Table_Helpers::Child>]’:
/usr/include/glibmm-2.4/glibmm/helperlist.h:70:16:   required from here
/usr/include/glibmm-2.4/glibmm/helperlist.h:73:13: warning: implicitly-declared ‘Glib::List_Iterator<Gtk::Table_Helpers::Child>& Glib::List_Iterator<Gtk::Table_Helpers::Child>::operator=(const Glib::List_Iterator<Gtk::Table_Helpers::Child>&)’ is deprecated [-Wdeprecated-copy]
   73 |       start = erase(start); // Implemented in derived class.
In file included from /usr/include/glibmm-2.4/glibmm/helperlist.h:28,
                 from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/project/mconstructor-main.cpp:22:
/usr/include/glibmm-2.4/glibmm/containers.h:78:3: note: because ‘Glib::List_Iterator<Gtk::Table_Helpers::Child>’ has user-provided ‘Glib::List_Iterator<T>::List_Iterator(const Self&) [with T = Gtk::Table_Helpers::Child; Glib::List_Iterator<T>::Self = Glib::List_Iterator<Gtk::Table_Helpers::Child>]’
   78 |   List_Iterator(const Self& src) : head_(src.head_), node_(src.node_) {}
      |   ^~~~~~~~~~~~~
In file included from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/project/mconstructor-main.cpp:22:
/usr/include/glibmm-2.4/glibmm/helperlist.h: In instantiation of ‘void Glib::HelperList<T_Child, T_CppElement, T_Iterator>::erase(Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator, Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator) [with T_Child = Gtk::ComboDropDownItem; T_CppElement = const Gtk::ComboDropDownItem; T_Iterator = Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>; Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator = Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>]’:
/usr/include/glibmm-2.4/glibmm/helperlist.h:70:16:   required from here
/usr/include/glibmm-2.4/glibmm/helperlist.h:73:13: warning: implicitly-declared ‘Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>& Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>::operator=(const Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>&)’ is deprecated [-Wdeprecated-copy]
   73 |       start = erase(start); // Implemented in derived class.
In file included from /usr/include/glibmm-2.4/glibmm/helperlist.h:28,
                 from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/project/mconstructor-main.cpp:22:
/usr/include/glibmm-2.4/glibmm/containers.h:192:3: note: because ‘Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>’ has user-provided ‘Glib::List_Cpp_Iterator<T_Impl, T_IFace>::List_Cpp_Iterator(const Self&) [with T_Impl = _GtkListItem; T_IFace = Gtk::ComboDropDownItem; Glib::List_Cpp_Iterator<T_Impl, T_IFace>::Self = Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>]’
  192 |   List_Cpp_Iterator(const Self& src) : head_(src.head_), node_(src.node_) {}
      |   ^~~~~~~~~~~~~~~~~
In file included from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/project/mconstructor-main.cpp:22:
/usr/include/glibmm-2.4/glibmm/helperlist.h: In instantiation of ‘void Glib::HelperList<T_Child, T_CppElement, T_Iterator>::erase(Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator, Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator) [with T_Child = Gtk::MenuItem; T_CppElement = const Gtk::Menu_Helpers::Element; T_Iterator = Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>; Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator = Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>]’:
/usr/include/glibmm-2.4/glibmm/helperlist.h:70:16:   required from here
/usr/include/glibmm-2.4/glibmm/helperlist.h:73:13: warning: implicitly-declared ‘Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>& Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>::operator=(const Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>&)’ is deprecated [-Wdeprecated-copy]
   73 |       start = erase(start); // Implemented in derived class.
In file included from /usr/include/glibmm-2.4/glibmm/helperlist.h:28,
                 from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/project/mconstructor-main.cpp:22:
/usr/include/glibmm-2.4/glibmm/containers.h:192:3: note: because ‘Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>’ has user-provided ‘Glib::List_Cpp_Iterator<T_Impl, T_IFace>::List_Cpp_Iterator(const Self&) [with T_Impl = _GtkMenuItem; T_IFace = Gtk::MenuItem; Glib::List_Cpp_Iterator<T_Impl, T_IFace>::Self = Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>]’
  192 |   List_Cpp_Iterator(const Self& src) : head_(src.head_), node_(src.node_) {}
      |   ^~~~~~~~~~~~~~~~~
In file included from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/project/mconstructor-main.cpp:22:
/usr/include/glibmm-2.4/glibmm/helperlist.h: In instantiation of ‘void Glib::HelperList<T_Child, T_CppElement, T_Iterator>::erase(Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator, Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator) [with T_Child = Gtk::Box_Helpers::Child; T_CppElement = const Gtk::Box_Helpers::Element; T_Iterator = Glib::List_Iterator<Gtk::Box_Helpers::Child>; Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator = Glib::List_Iterator<Gtk::Box_Helpers::Child>]’:
/usr/include/glibmm-2.4/glibmm/helperlist.h:70:16:   required from here
/usr/include/glibmm-2.4/glibmm/helperlist.h:73:13: warning: implicitly-declared ‘Glib::List_Iterator<Gtk::Box_Helpers::Child>& Glib::List_Iterator<Gtk::Box_Helpers::Child>::operator=(const Glib::List_Iterator<Gtk::Box_Helpers::Child>&)’ is deprecated [-Wdeprecated-copy]
   73 |       start = erase(start); // Implemented in derived class.
In file included from /usr/include/glibmm-2.4/glibmm/helperlist.h:28,
                 from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/project/mconstructor-main.cpp:22:
/usr/include/glibmm-2.4/glibmm/containers.h:78:3: note: because ‘Glib::List_Iterator<Gtk::Box_Helpers::Child>’ has user-provided ‘Glib::List_Iterator<T>::List_Iterator(const Self&) [with T = Gtk::Box_Helpers::Child; Glib::List_Iterator<T>::Self = Glib::List_Iterator<Gtk::Box_Helpers::Child>]’
   78 |   List_Iterator(const Self& src) : head_(src.head_), node_(src.node_) {}
      |   ^~~~~~~~~~~~~
g++ -o build/src/mgui/dialog.o -c -Wall -W -Wno-reorder -O2 -ansi -std=c++11 -Wno-deprecated-declarations -DBOOST_SP_USE_QUICK_ALLOCATOR -DBOOST_SYSTEM_NO_DEPRECATED -DNDEBUG -DEXT_BOOST -Ibuild/src -Isrc -Ilibs/boost-logging -Ilibs/asl -Ibuild/src -Isrc -I/usr/include/glibmm-2.4 -I/usr/lib/x86_64-linux-gnu/glibmm-2.4/include -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/sigc++-2.0 -I/usr/lib/x86_64-linux-gnu/sigc++-2.0/include -I/usr/include/libxml++-2.6 -I/usr/lib/x86_64-linux-gnu/libxml++-2.6/include -I/usr/include/libxml2 -I/usr/include/x86_64-linux-gnu -I/usr/include/gtkmm-2.4 -I/usr/lib/x86_64-linux-gnu/gtkmm-2.4/include -I/usr/include/atkmm-1.6 -I/usr/include/gtk-unix-print-2.0 -I/usr/include/gtk-2.0 -I/usr/include/gdkmm-2.4 -I/usr/lib/x86_64-linux-gnu/gdkmm-2.4/include -I/usr/include/giomm-2.4 -I/usr/lib/x86_64-linux-gnu/giomm-2.4/include -I/usr/include/pangomm-1.4 -I/usr/lib/x86_64-linux-gnu/pangomm-1.4/include -I/usr/include/cairomm-1.0 -I/usr/lib/x86_64-linux-gnu/cairomm-1.0/include -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/harfbuzz -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/libpng16 src/mgui/dialog.cpp
In file included from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dialog.cpp:22:
/usr/include/glibmm-2.4/glibmm/helperlist.h: In instantiation of ‘void Glib::HelperList<T_Child, T_CppElement, T_Iterator>::erase(Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator, Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator) [with T_Child = Gtk::Table_Helpers::Child; T_CppElement = Gtk::Widget; T_Iterator = Glib::List_Iterator<Gtk::Table_Helpers::Child>; Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator = Glib::List_Iterator<Gtk::Table_Helpers::Child>]’:
/usr/include/glibmm-2.4/glibmm/helperlist.h:70:16:   required from here
/usr/include/glibmm-2.4/glibmm/helperlist.h:73:13: warning: implicitly-declared ‘Glib::List_Iterator<Gtk::Table_Helpers::Child>& Glib::List_Iterator<Gtk::Table_Helpers::Child>::operator=(const Glib::List_Iterator<Gtk::Table_Helpers::Child>&)’ is deprecated [-Wdeprecated-copy]
   73 |       start = erase(start); // Implemented in derived class.
In file included from /usr/include/glibmm-2.4/glibmm/helperlist.h:28,
                 from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dialog.cpp:22:
/usr/include/glibmm-2.4/glibmm/containers.h:78:3: note: because ‘Glib::List_Iterator<Gtk::Table_Helpers::Child>’ has user-provided ‘Glib::List_Iterator<T>::List_Iterator(const Self&) [with T = Gtk::Table_Helpers::Child; Glib::List_Iterator<T>::Self = Glib::List_Iterator<Gtk::Table_Helpers::Child>]’
   78 |   List_Iterator(const Self& src) : head_(src.head_), node_(src.node_) {}
      |   ^~~~~~~~~~~~~
In file included from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dialog.cpp:22:
/usr/include/glibmm-2.4/glibmm/helperlist.h: In instantiation of ‘void Glib::HelperList<T_Child, T_CppElement, T_Iterator>::erase(Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator, Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator) [with T_Child = Gtk::ComboDropDownItem; T_CppElement = const Gtk::ComboDropDownItem; T_Iterator = Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>; Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator = Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>]’:
/usr/include/glibmm-2.4/glibmm/helperlist.h:70:16:   required from here
/usr/include/glibmm-2.4/glibmm/helperlist.h:73:13: warning: implicitly-declared ‘Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>& Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>::operator=(const Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>&)’ is deprecated [-Wdeprecated-copy]
   73 |       start = erase(start); // Implemented in derived class.
In file included from /usr/include/glibmm-2.4/glibmm/helperlist.h:28,
                 from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dialog.cpp:22:
/usr/include/glibmm-2.4/glibmm/containers.h:192:3: note: because ‘Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>’ has user-provided ‘Glib::List_Cpp_Iterator<T_Impl, T_IFace>::List_Cpp_Iterator(const Self&) [with T_Impl = _GtkListItem; T_IFace = Gtk::ComboDropDownItem; Glib::List_Cpp_Iterator<T_Impl, T_IFace>::Self = Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>]’
  192 |   List_Cpp_Iterator(const Self& src) : head_(src.head_), node_(src.node_) {}
      |   ^~~~~~~~~~~~~~~~~
In file included from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dialog.cpp:22:
/usr/include/glibmm-2.4/glibmm/helperlist.h: In instantiation of ‘void Glib::HelperList<T_Child, T_CppElement, T_Iterator>::erase(Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator, Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator) [with T_Child = Gtk::MenuItem; T_CppElement = const Gtk::Menu_Helpers::Element; T_Iterator = Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>; Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator = Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>]’:
/usr/include/glibmm-2.4/glibmm/helperlist.h:70:16:   required from here
/usr/include/glibmm-2.4/glibmm/helperlist.h:73:13: warning: implicitly-declared ‘Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>& Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>::operator=(const Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>&)’ is deprecated [-Wdeprecated-copy]
   73 |       start = erase(start); // Implemented in derived class.
In file included from /usr/include/glibmm-2.4/glibmm/helperlist.h:28,
                 from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dialog.cpp:22:
/usr/include/glibmm-2.4/glibmm/containers.h:192:3: note: because ‘Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>’ has user-provided ‘Glib::List_Cpp_Iterator<T_Impl, T_IFace>::List_Cpp_Iterator(const Self&) [with T_Impl = _GtkMenuItem; T_IFace = Gtk::MenuItem; Glib::List_Cpp_Iterator<T_Impl, T_IFace>::Self = Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>]’
  192 |   List_Cpp_Iterator(const Self& src) : head_(src.head_), node_(src.node_) {}
      |   ^~~~~~~~~~~~~~~~~
In file included from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dialog.cpp:22:
/usr/include/glibmm-2.4/glibmm/helperlist.h: In instantiation of ‘void Glib::HelperList<T_Child, T_CppElement, T_Iterator>::erase(Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator, Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator) [with T_Child = Gtk::Box_Helpers::Child; T_CppElement = const Gtk::Box_Helpers::Element; T_Iterator = Glib::List_Iterator<Gtk::Box_Helpers::Child>; Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator = Glib::List_Iterator<Gtk::Box_Helpers::Child>]’:
/usr/include/glibmm-2.4/glibmm/helperlist.h:70:16:   required from here
/usr/include/glibmm-2.4/glibmm/helperlist.h:73:13: warning: implicitly-declared ‘Glib::List_Iterator<Gtk::Box_Helpers::Child>& Glib::List_Iterator<Gtk::Box_Helpers::Child>::operator=(const Glib::List_Iterator<Gtk::Box_Helpers::Child>&)’ is deprecated [-Wdeprecated-copy]
   73 |       start = erase(start); // Implemented in derived class.
In file included from /usr/include/glibmm-2.4/glibmm/helperlist.h:28,
                 from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dialog.cpp:22:
/usr/include/glibmm-2.4/glibmm/containers.h:78:3: note: because ‘Glib::List_Iterator<Gtk::Box_Helpers::Child>’ has user-provided ‘Glib::List_Iterator<T>::List_Iterator(const Self&) [with T = Gtk::Box_Helpers::Child; Glib::List_Iterator<T>::Self = Glib::List_Iterator<Gtk::Box_Helpers::Child>]’
   78 |   List_Iterator(const Self& src) : head_(src.head_), node_(src.node_) {}
      |   ^~~~~~~~~~~~~
g++ -o build/src/mgui/dvdimport.o -c -Wall -W -Wno-reorder -O2 -ansi -std=c++11 -Wno-deprecated-declarations -DBOOST_SP_USE_QUICK_ALLOCATOR -DBOOST_SYSTEM_NO_DEPRECATED -DNDEBUG -DEXT_BOOST -Ibuild/src -Isrc -Ilibs/boost-logging -Ilibs/asl -Ibuild/src -Isrc -I/usr/include/glibmm-2.4 -I/usr/lib/x86_64-linux-gnu/glibmm-2.4/include -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/sigc++-2.0 -I/usr/lib/x86_64-linux-gnu/sigc++-2.0/include -I/usr/include/libxml++-2.6 -I/usr/lib/x86_64-linux-gnu/libxml++-2.6/include -I/usr/include/libxml2 -I/usr/include/x86_64-linux-gnu -I/usr/include/gtkmm-2.4 -I/usr/lib/x86_64-linux-gnu/gtkmm-2.4/include -I/usr/include/atkmm-1.6 -I/usr/include/gtk-unix-print-2.0 -I/usr/include/gtk-2.0 -I/usr/include/gdkmm-2.4 -I/usr/lib/x86_64-linux-gnu/gdkmm-2.4/include -I/usr/include/giomm-2.4 -I/usr/lib/x86_64-linux-gnu/giomm-2.4/include -I/usr/include/pangomm-1.4 -I/usr/lib/x86_64-linux-gnu/pangomm-1.4/include -I/usr/include/cairomm-1.0 -I/usr/lib/x86_64-linux-gnu/cairomm-1.0/include -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/harfbuzz -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/libpng16 src/mgui/dvdimport.cpp
In file included from src/mgui/dvdimport.cpp:38:
src/mbase/project/table.h:131:73: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
  131 |                             const std::string& cur_dir = std::string()) throw (std::exception);
      |                                                                         ^~~~~
src/mgui/dvdimport.cpp: In function ‘void DVD::OnPreparePage(DVD::ImportData&)’:
src/mgui/dvdimport.cpp:214:82: error: ambiguous overload for ‘operator<<’ (operand types are ‘str::stream’ and ‘const char [3]’)
  214 |                 std::string desc = (str::stream(Mpeg::SecToHMS(vob.tmLen, true)) <<  ", "
      |                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~  ~~~~
      |                                          |                                           |
      |                                          str::stream                                 const char [3]
In file included from src/mgui/project/add.h:28,
                 from src/mgui/dvdimport.cpp:27:
src/mlib/string.h:54:18: note: candidate: ‘str::stream& str::stream::operator<<(T) [with T = const char*]’
   54 |          stream& operator << (T t)
      |                  ^~~~~~~~
In file included from /usr/include/boost/format/detail/compat_workarounds.hpp:60,
                 from /usr/include/boost/format.hpp:27,
                 from src/mlib/format.h:49,
                 from src/mbase/project/_extlibs_.h:37,
                 from src/mgui/_extlibs_.h:31,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dvdimport.cpp:22:
/usr/include/c++/9/ostream:691:5: note: candidate: ‘typename std::enable_if<std::__and_<std::__not_<std::is_lvalue_reference<_Tp> >, std::__is_convertible_to_basic_ostream<_Ostream>, std::__is_insertable<typename std::__is_convertible_to_basic_ostream<_Tp>::__ostream_type, const _Tp&, void> >::value, typename std::__is_convertible_to_basic_ostream<_Tp>::__ostream_type>::type std::operator<<(_Ostream&&, const _Tp&) [with _Ostream = str::stream; _Tp = char [3]; typename std::enable_if<std::__and_<std::__not_<std::is_lvalue_reference<_Tp> >, std::__is_convertible_to_basic_ostream<_Ostream>, std::__is_insertable<typename std::__is_convertible_to_basic_ostream<_Tp>::__ostream_type, const _Tp&, void> >::value, typename std::__is_convertible_to_basic_ostream<_Tp>::__ostream_type>::type = std::basic_ostream<char>&]’
  691 |     operator<<(_Ostream&& __os, const _Tp& __x)
      |     ^~~~~~~~
In file included from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dvdimport.cpp:22:
/usr/include/glibmm-2.4/glibmm/helperlist.h: In instantiation of ‘void Glib::HelperList<T_Child, T_CppElement, T_Iterator>::erase(Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator, Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator) [with T_Child = Gtk::Table_Helpers::Child; T_CppElement = Gtk::Widget; T_Iterator = Glib::List_Iterator<Gtk::Table_Helpers::Child>; Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator = Glib::List_Iterator<Gtk::Table_Helpers::Child>]’:
/usr/include/glibmm-2.4/glibmm/helperlist.h:70:16:   required from here
/usr/include/glibmm-2.4/glibmm/helperlist.h:73:13: warning: implicitly-declared ‘Glib::List_Iterator<Gtk::Table_Helpers::Child>& Glib::List_Iterator<Gtk::Table_Helpers::Child>::operator=(const Glib::List_Iterator<Gtk::Table_Helpers::Child>&)’ is deprecated [-Wdeprecated-copy]
   73 |       start = erase(start); // Implemented in derived class.
In file included from /usr/include/glibmm-2.4/glibmm/helperlist.h:28,
                 from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dvdimport.cpp:22:
/usr/include/glibmm-2.4/glibmm/containers.h:78:3: note: because ‘Glib::List_Iterator<Gtk::Table_Helpers::Child>’ has user-provided ‘Glib::List_Iterator<T>::List_Iterator(const Self&) [with T = Gtk::Table_Helpers::Child; Glib::List_Iterator<T>::Self = Glib::List_Iterator<Gtk::Table_Helpers::Child>]’
   78 |   List_Iterator(const Self& src) : head_(src.head_), node_(src.node_) {}
      |   ^~~~~~~~~~~~~
In file included from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dvdimport.cpp:22:
/usr/include/glibmm-2.4/glibmm/helperlist.h: In instantiation of ‘void Glib::HelperList<T_Child, T_CppElement, T_Iterator>::erase(Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator, Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator) [with T_Child = Gtk::ComboDropDownItem; T_CppElement = const Gtk::ComboDropDownItem; T_Iterator = Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>; Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator = Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>]’:
/usr/include/glibmm-2.4/glibmm/helperlist.h:70:16:   required from here
/usr/include/glibmm-2.4/glibmm/helperlist.h:73:13: warning: implicitly-declared ‘Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>& Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>::operator=(const Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>&)’ is deprecated [-Wdeprecated-copy]
   73 |       start = erase(start); // Implemented in derived class.
In file included from /usr/include/glibmm-2.4/glibmm/helperlist.h:28,
                 from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dvdimport.cpp:22:
/usr/include/glibmm-2.4/glibmm/containers.h:192:3: note: because ‘Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>’ has user-provided ‘Glib::List_Cpp_Iterator<T_Impl, T_IFace>::List_Cpp_Iterator(const Self&) [with T_Impl = _GtkListItem; T_IFace = Gtk::ComboDropDownItem; Glib::List_Cpp_Iterator<T_Impl, T_IFace>::Self = Glib::List_Cpp_Iterator<_GtkListItem, Gtk::ComboDropDownItem>]’
  192 |   List_Cpp_Iterator(const Self& src) : head_(src.head_), node_(src.node_) {}
      |   ^~~~~~~~~~~~~~~~~
In file included from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dvdimport.cpp:22:
/usr/include/glibmm-2.4/glibmm/helperlist.h: In instantiation of ‘void Glib::HelperList<T_Child, T_CppElement, T_Iterator>::erase(Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator, Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator) [with T_Child = Gtk::MenuItem; T_CppElement = const Gtk::Menu_Helpers::Element; T_Iterator = Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>; Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator = Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>]’:
/usr/include/glibmm-2.4/glibmm/helperlist.h:70:16:   required from here
/usr/include/glibmm-2.4/glibmm/helperlist.h:73:13: warning: implicitly-declared ‘Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>& Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>::operator=(const Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>&)’ is deprecated [-Wdeprecated-copy]
   73 |       start = erase(start); // Implemented in derived class.
In file included from /usr/include/glibmm-2.4/glibmm/helperlist.h:28,
                 from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dvdimport.cpp:22:
/usr/include/glibmm-2.4/glibmm/containers.h:192:3: note: because ‘Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>’ has user-provided ‘Glib::List_Cpp_Iterator<T_Impl, T_IFace>::List_Cpp_Iterator(const Self&) [with T_Impl = _GtkMenuItem; T_IFace = Gtk::MenuItem; Glib::List_Cpp_Iterator<T_Impl, T_IFace>::Self = Glib::List_Cpp_Iterator<_GtkMenuItem, Gtk::MenuItem>]’
  192 |   List_Cpp_Iterator(const Self& src) : head_(src.head_), node_(src.node_) {}
      |   ^~~~~~~~~~~~~~~~~
In file included from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dvdimport.cpp:22:
/usr/include/glibmm-2.4/glibmm/helperlist.h: In instantiation of ‘void Glib::HelperList<T_Child, T_CppElement, T_Iterator>::erase(Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator, Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator) [with T_Child = Gtk::Box_Helpers::Child; T_CppElement = const Gtk::Box_Helpers::Element; T_Iterator = Glib::List_Iterator<Gtk::Box_Helpers::Child>; Glib::HelperList<T_Child, T_CppElement, T_Iterator>::iterator = Glib::List_Iterator<Gtk::Box_Helpers::Child>]’:
/usr/include/glibmm-2.4/glibmm/helperlist.h:70:16:   required from here
/usr/include/glibmm-2.4/glibmm/helperlist.h:73:13: warning: implicitly-declared ‘Glib::List_Iterator<Gtk::Box_Helpers::Child>& Glib::List_Iterator<Gtk::Box_Helpers::Child>::operator=(const Glib::List_Iterator<Gtk::Box_Helpers::Child>&)’ is deprecated [-Wdeprecated-copy]
   73 |       start = erase(start); // Implemented in derived class.
In file included from /usr/include/glibmm-2.4/glibmm/helperlist.h:28,
                 from /usr/include/glibmm-2.4/glibmm.h:113,
                 from /usr/include/gtkmm-2.4/gtkmm.h:87,
                 from src/mgui/_extlibs_.h:46,
                 from src/mgui/_pc_.h:29,
                 from src/mgui/dvdimport.cpp:22:
/usr/include/glibmm-2.4/glibmm/containers.h:78:3: note: because ‘Glib::List_Iterator<Gtk::Box_Helpers::Child>’ has user-provided ‘Glib::List_Iterator<T>::List_Iterator(const Self&) [with T = Gtk::Box_Helpers::Child; Glib::List_Iterator<T>::Self = Glib::List_Iterator<Gtk::Box_Helpers::Child>]’
   78 |   List_Iterator(const Self& src) : head_(src.head_), node_(src.node_) {}
      |   ^~~~~~~~~~~~~
scons: *** [build/src/mgui/dvdimport.o] Error 1
scons: building terminated because of errors.

**Comment by muravjov** [comment]
Hello @roracle ,
I've just invited you to https://github.com/bombono-dvd organisation, you are to receive the email shortly. Feel free to apply the patches to the code and merge.

Thank you for your work.

**Comment by roracle** [comment]
> Hello @roracle ,
> I've just invited you to https://github.com/bombono-dvd organisation, you are to receive the email shortly. Feel free to apply the patches to the code and merge.
> 
> Thank you for your work.

Not a problem.  I'm not too sure about how to submit the files, or which files I changed.  It was a LOT of changes and I did it on the fly, to be honest.  I could always do it again, keep inventory, then submit the files over.

I'm still experiencing a build problem, and I think it has to do with python2 code and it needing to be python3, but I'm not that well versed in all of this.

The issue is when it tries to make the config files (cfg_file).  I think the algorighm changed with python3 from what was originally in the code.

**Comment by roracle** [comment]
UPDATE:
I GOT IT!  Only thing is, I'm using the files coming from the official Bombono website, not sure if this master branch is the same.

I have yet to test it out, but I intend on doing so later (it's almost 4 AM here, and I need sleep).

The only problem is I don't know how to create packages.  The best I can do is go through the changes with the master branch and make the changes and submit them.

I'm just tickled that I was able to get this working.  It's a must-have application for me, as I do a side business of converting VHS files to DVD discs, and Bombono is the only one (even among the premium software out there) that can do video clips in the menu itself.

Anyways, I'm tired, and I'm going to bed.  We'll work on this more later!

**Comment by ghoppaccia** [comment]
@roracle I'm following this. I really need this app and can't make it work in 20.04. Tried to package it as Appimage but it wouldn't work. How did you manage to compile it?
As of now I have a issue with libdvdread.
Thanks!

**Comment by roracle** [comment]
> @roracle I'm following this. I really need this app and can't make it work in 20.04. Tried to package it as Appimage but it wouldn't work. How did you manage to compile it?
> As of now I have a issue with libdvdread.
> Thanks!

I found some diff patch files from whoever is maintaining it on Suse.  I made the changes to the code, commented out a package statement, and got it compiled.

Right now I'm installing Ubuntu 20.04 on VirtualBox and I'll be testing the compiling very soon with the master branch and the changes I made to it.

My goal is to get it packaged up for 20.04 soon.  Stay tuned!

**Comment by ghoppaccia** [comment]
> > @roracle I'm following this. I really need this app and can't make it work in 20.04. Tried to package it as Appimage but it wouldn't work. How did you manage to compile it?
> > As of now I have a issue with libdvdread.
> > Thanks!
> 
> I found some diff patch files from whoever is maintaining it on Suse. I made the changes to the code, commented out a package statement, and got it compiled.
> 
> Right now I'm installing Ubuntu 20.04 on VirtualBox and I'll be testing the compiling very soon with the master branch and the changes I made to it.
> 
> My goal is to get it packaged up for 20.04 soon. Stay tuned!

Oh thank you so much, that would be amazing! This has given me a lot of headache trying to make it work in 20.04!

**Comment by roracle** [comment]
So I don't know what's different between what I installed on my own system and what I installed on the VirtualBox system, but it doesn't want to compile at all.  I might package it up from my own system, but I've never actually built a deb file before.  This might take a while.

**Comment by ghoppaccia** [comment]
> So I don't know what's different between what I installed on my own system and what I installed on the VirtualBox system, but it doesn't want to compile at all. I might package it up from my own system, but I've never actually built a deb file before. This might take a while.

If you tell me what you did to compile it on your system I could try on mine. Yesterday I managed to advance up to a certain point, but then I got stuck again.

**Comment by roracle** [comment]
Okay...
1)I installed all the dev and regular libs as suggested on the README on the main page of this Git.
2)Next, I downloaded 1.2.4 from the Bombono Git using the provided link [here](https://github.com/bombono-dvd/bombono-dvd/archive/1.2.4.tar.gz).  Extract these files.
3) Then I went through all the files on [this page](https://pmbs.links2linux.org/package/show/SLE12/bombono) with the "patch" extension and manually edited EVERYTHING becasue I wasn't sure of two things: how to use diff patch files, and unsure if it was compatible with any other changes made by the person building for Suse.  I'm assuming they took the same sources, but I don't know for sure.
4) Run scons
5) Run sudo scons install

**Comment by roracle** [comment]
I just realized, it's probably a bunch of dev packages you need to install.
The list on the README on the main page tells you what it depends on, yet there are other things it's going to ask when you try to compile.
For example, it wants various boost packages, but it's not enough to install the binaries, but the libboostXX-dev packages as well.
I'll be trying again tomorrow, but for now I'm just figuring it out from a reinstall of my OS on my main drive.
Look for the errors it throws.  If you can't figure out which package to install, just ask me and I'll get to you as soon as I can.

**Comment by ghoppaccia** [comment]
> I just realized, it's probably a bunch of dev packages you need to install.
> The list on the README on the main page tells you what it depends on, yet there are other things it's going to ask when you try to compile.
> For example, it wants various boost packages, but it's not enough to install the binaries, but the libboostXX-dev packages as well.
> I'll be trying again tomorrow, but for now I'm just figuring it out from a reinstall of my OS on my main drive.
> Look for the errors it throws. If you can't figure out which package to install, just ask me and I'll get to you as soon as I can.

Well well, I have to say I'm surprised this went easier than I thought. Had some issues with the dependencies at the beginning, but I managed to fix those and the building went pretty smooth I have to say. I used the zip file you provided.
I still have to check if the program is actually working or if it has some other issue, but so far it is starting with no errors and seem to be fine.
Thank you so much!

Now I have to see if I manage to package it in some way to be able to install it on other machines and make it available to other people...

**Comment by roracle** [comment]
I think I got that exact same error, but I've slept since then so don't quote me on that lol.  I might be an advanced user, but I have never messed with packaging before.  But I'm willing to learn if it menas having a simple install for this software.

**Comment by ghoppaccia** [comment]
> I think I got that exact same error, but I've slept since then so don't quote me on that lol. I might be an advanced user, but I have never messed with packaging before. But I'm willing to learn if it menas having a simple install for this software.

Sorry I forgot to attach the deb file, in case you might want to take a look. I don't know if I can find the AppImage I made, I deleted it because of frustration on that, but the error was still the same.

bombono-dvd.deb:
[https://covid19.framadrop.org/lufi/r/ZmK5Pv1EtT#qXYF/rxbOuiMVxLosGynnAveVzKWUOCL1HGKhSVb0yw=](url)

And I used it with the 1.2.2. bombono-dvd-data.deb which you can find anywhere, but in case you want it it's here:
[https://covid19.framadrop.org/lufi/r/Bvqf2ZRJQ8#w0lmMEhrphH+O16svvqgbtNO1Qg+bIe+TRH1nfWHDIg=](url)

By the way I'll let you know when I'll make a test but yes, the software is great. I am so thankful to you that, even if building from source is no the best way to install apps, at least I could make it work on my machine!

**Comment by roracle** [comment]
Awesome! Thanks a lot for that, I'll take a look at it and see what I can accomplish.
I just finished compiling on the VirtualBox I set up.  I've been trying to keep track of everything people would need to get it compiled, but at the same time, the ultimate goal should be to get a deb package built and working.
If you get a deb working before I do, just post it on here so we can keep updated!

**Comment by ghoppaccia** [comment]
Definitely! I'll keep you updated. Anyway I'm no expert so I might be doing something wrong, or at least not properly, sorry for that!
By the way I just tried to burn a video DVD and it didn't seem to work, I will try to troubleshoot. It might be my fault since before starting burning it gave me the message that the size would exceed the DVD. Usually is not a big issue but I will look into that.

**Comment by roracle** [comment]
> Definitely! I'll keep you updated. Anyway I'm no expert in this, I'm a Linux user since more than 10 years, but never got into programming or packaging apps, so I might be doing something wrong, or at least not properly, sorry for that!
> By the way I just tried to burn a video DVD and it didn't seem to work, I will try to troubleshoot. It might be my fault since before starting burning it gave me the message that the size would exceed the DVD. It is usually not a issue but I will look into that.
> 
> EDIT: tried to burn the DVD making the menu this time (I usually don't do it) and it does work. The menu appears but just a sec and then goes away. But then playing PLAY (on my DVD player) it works. I don't know what might be the cause, I saved the Bombono log, but I won't post it here since is not relevant with the issue. I will make more trials, anyway all in all it is working, I got the video DVD playing so probably I just have to fickle with some setting. Hurrà! Thank you so much!
> In the next days I will look into trying to package it in some way to have it available for other people and systems.

I haven't actually burned a disc yet.  I don't do that on this particular computer anyway, I just make the ISO on this and burn it on another. But from what I know, any ISO I made before 20.04 always burned and played properly.  Of course, I was always careful to ensure it was in the right format and under the maximum size.  That can be an issue if done improperly.
But again, I'm glad it's working for what it's worth.
I'm not sure why this project insists on sticking with python2 code at the base.  Perhaps I should branch it off and create a new repo that is Ubuntu-specific, to keep up to date with the newer code.
I could build an install script that will install all required packages, compile, and install.  This would all be done from source.
Makes me wonder why Ubuntu doesn't have something like the AUR which does the same thing.  Oh well, one thing at a time, I suppose.
(BTW: the reason I don't use Manjaro is because of my hybrid Nvidia/Intel video chip, otherwise I might have never come here and been part of this little project we've found ourselves working on lol)

**Comment by ghoppaccia** [comment]
> > Definitely! I'll keep you updated. Anyway I'm no expert in this, I'm a Linux user since more than 10 years, but never got into programming or packaging apps, so I might be doing something wrong, or at least not properly, sorry for that!
> > By the way I just tried to burn a video DVD and it didn't seem to work, I will try to troubleshoot. It might be my fault since before starting burning it gave me the message that the size would exceed the DVD. It is usually not a issue but I will look into that.
> > EDIT: tried to burn the DVD making the menu this time (I usually don't do it) and it does work. The menu appears but just a sec and then goes away. But then playing PLAY (on my DVD player) it works. I don't know what might be the cause, I saved the Bombono log, but I won't post it here since is not relevant with the issue. I will make more trials, anyway all in all it is working, I got the video DVD playing so probably I just have to fickle with some setting. Hurrà! Thank you so much!
> > In the next days I will look into trying to package it in some way to have it available for other people and systems.
> 
> I haven't actually burned a disc yet. I don't do that on this particular computer anyway, I just make the ISO on this and burn it on another. But from what I know, any ISO I made before 20.04 always burned and played properly. Of course, I was always careful to ensure it was in the right format and under the maximum size. That can be an issue if done improperly.
> But again, I'm glad it's working for what it's worth.
> I'm not sure why this project insists on sticking with python2 code at the base. Perhaps I should branch it off and create a new repo that is Ubuntu-specific, to keep up to date with the newer code.
> I could build an install script that will install all required packages, compile, and install. This would all be done from source.
> Makes me wonder why Ubuntu doesn't have something like the AUR which does the same thing. Oh well, one thing at a time, I suppose.
> (BTW: the reason I don't use Manjaro is because of my hybrid Nvidia/Intel video chip, otherwise I might have never come here and been part of this little project we've found ourselves working on lol)

Ahah, it's funny since I was about to install Manjaro on a second Hard disk, to fickle around a bit, and I saw that they have Bombono in AUR, so I thought I could use it for that! (But then I found your help!)

Anyway a quick update on my issue, it seems that if I directly burn the DVD, without any menu, it doesn't work properly. With the menu it kind of does, the menu disappears after a second, but if you press play the video starts.
BUT if I create an .iso file and then burn it with the stock ubuntu app it works flawlessly.
Now, I don't know from what it could depend, but working in this way is not a problem at all for me.
Just to keep in mind that the burning feature has some issue, at least for me.

**Comment by roracle** [comment]
> 
> Ahah, it's funny since I was about to install Manjaro on a second Hard disk, to fickle around a bit, and I saw that they have Bombono in AUR, so I thought I could use it for that! (But then I found your help!)
> 
> Anyway a quick update on my issue, it seems that if I directly burn the DVD, without any menu, it doesn't work. With the menu it kind of does, the menu disappears after a second, but if you press play the video starts.
> BUT if I create an .iso file and then burn it with the stock ubuntu app it works flawlessly.
> Now, I don't know what might be the cause, but working in this way is not a problem at all for me.
> Just to keep in mind that the burning feature has some issue, at least for me.

*deletes excess quotes*

Out of curiosity, does it make an ISO when you do a direct burn?  I ask becasue the way I do things, I keep an ISO as a backup of all projects in case customers want to puchase extra copies in the future (at a discount, obviously, as the hard part of recording and creating is already done, they're basically paying for a disc and the time burning it).  But yeah, lots of people do discs for gifts.  Always good to see happy people enjoying their old home movies, knowing they'll be preserved. :)

**Comment by ghoppaccia** [comment]
> Out of curiosity, does it make an ISO when you do a direct burn? I ask becasue the way I do things, I keep an ISO as a backup of all projects in case customers want to puchase extra copies in the future (at a discount, obviously, as the hard part of recording and creating is already done, they're basically paying for a disc and the time burning it). But yeah, lots of people do discs for gifts. Always good to see happy people enjoying their old home movies, knowing they'll be preserved. :)

Eheh, that's what my dad does, for its own VHSes and DVDs library!

Anyway, I have checked and it doesn't. It creates the VIDEO_TS and AUDIO_TS folder structure (in its cache folder) and then burns it directly, in some way. I imagine that might be where the issue lies, the way of burning it like that. Anyway I think it's strange since obviously the .iso it creates is simply that same folder structure inside a .iso file...

**Comment by Doranwen** [comment]
@echeoquehaii 
> https://covid19.framadrop.org/lufi/r/ZmK5Pv1EtT#qXYF/rxbOuiMVxLosGynnAveVzKWUOCL1HGKhSVb0yw=

The links to the deb files don't work.  Do you still have working deb files for this program?  I upgraded to Mint 21.2 (based on Ubuntu 20.04) and was sorely disappointed to find Bombono was no longer available.

**Comment by muravjov** [comment]
> 
> The links to the deb files don't work. Do you still have working deb files for this program? I upgraded to Mint 21.2 (based on Ubuntu 20.04) and was sorely disappointed to find Bombono was no longer available.

I am not aware whether bombono is built for the mint 21.2 .
Maybe someone will help to build packages.

**Comment by echeoquehaii** [comment]
> The links to the deb files don't work. Do you still have working deb files for this program? I upgraded to Mint 21.2 (based on Ubuntu 20.04) and was sorely disappointed to find Bombono was no longer available.

Hey @Doranwen! I just searched on my disks and backups and unfortunately I found no trace of the file. I have migrated to Manjaro since quite a while and probably deleted it, I'm sorry!

I have no time to try to build it again now, also considering that I don't have Ubuntu on any of my machines, but I suggest trying yourself. If I succeeded as an amateur, I'm sure you can too ;) Start from [here](https://github.com/bombono-dvd/bombono-dvd/issues/25#issuecomment-623676710)

Or maybe @roracle or someone else in this thread still have a copy of it.

Good luck! 

PS: Also found [this](https://forums.linuxmint.com/viewtopic.php?t=338703) which points [here](https://software.opensuse.org//download.html?project=home%3Atkb&package=bombono-dvd). Maybe it helps ;)
